### DISCUSSION TITLE
Does Gramine-TDX support shared memory between TEE VM and host OS?

### DISCUSSION THREAD

**Discussion by ziqi-zhang** [gramineproject/gramine#2078] [General]
Hi, I wonder if there is any memory-sharing mechanism that allows grime-TDX to share memory between the VM and host OS. My goal is to efficiently transfer data between the VM and the host OS.

**Comment by monavij** [comment]
Gramine-TDX supports virtIo for communication with the host that uses shared memory virtio queues. . It supports a vsock driver for network communication and a virtual FS driver for FS access, as it does not have its own network or FS stack. You can find more details in our paper - https://www.sigsac.org/ccs/CCS2024/assets/pfaubmkaccs2024submissions/new/p4598-kuvaiskii.pdf
